### ISSUE TITLE
F3D 2.4.0 Roadmap

### ISSUE THREAD

**Issue by mwestphal** [f3d-app/f3d#1242]
**Context**
F3D is a collaborative project, let's collaboratively construct the roadmap for 2.4.0 release!

**Suggested solution**
Comment and discuss below, maintainers will take suggestions into account to add to the roadmap.
Please refer to existing issue when adding comments. Feel free to create issues if your ideas are not in issues already.

**Current roadmap**

TODO

**Comment by mwestphal** [comment]
A few different areas of improvements imo.

1. Plugin framework consolidation

Our plugin framework is working very well, but it is missing a few features
 - https://github.com/f3d-app/f3d/issues/1135
 - https://github.com/f3d-app/f3d/issues/1205
 - https://github.com/f3d-app/f3d/issues/1115
 - https://github.com/f3d-app/f3d/issues/1193
 
3. Format consolidation

We have added many file format support in the last few minor releases, let's consolidate them:
 - https://github.com/f3d-app/f3d/issues/1204
 - https://github.com/f3d-app/f3d/issues/1187
 - https://github.com/f3d-app/f3d/issues/1178
 - https://github.com/f3d-app/f3d/issues/1139
 - https://github.com/f3d-app/f3d/issues/913
 - https://github.com/f3d-app/f3d/issues/469
 
 (not all issues listed here, only the one the seems the most important to me)

5. Desktop integration improvements

Desktop integration has been on the backburner for a while, it would be nice to improve that in the coming months.
We may need to find experts of the respective OSes though:

 - Windows meta issue: https://github.com/f3d-app/f3d/issues/21
 - macOS meta issue: https://github.com/f3d-app/f3d/issues/775
 - https://github.com/f3d-app/f3d/issues/1101
 - https://github.com/f3d-app/f3d/issues/1100
 
7. Window types and interactions

Recent work by @snoyer as well as input from @GitCodeBoy made us realize that we may want to provide more in terms of interoparabilty with window types and interactions. It is more of a discussion that actual issues to work on but this should move forward:

 - https://github.com/f3d-app/f3d/issues/418
 - https://github.com/f3d-app/f3d/issues/1246

9. Documentation efforts

Our doc system is reliable but it is lacking in many ways, improving it would go a long way:

 - #1218 
 - https://github.com/f3d-app/f3d/issues/845
 - https://github.com/f3d-app/f3d/issues/590
 
11. QoL features

A few, highly requested features that stand out to me:

 - #27 
 - #86 
 - https://github.com/f3d-app/f3d/issues/25
 - https://github.com/f3d-app/f3d/issues/119

13. Any other help-wanted bug fix or feature

Any issues or features that is labeled with "help-wanted" could be part of 2.4.0 release

**Comment by Meakk** [comment]
Wow, this is a great sum up! Nice @mwestphal!  
Should we add new features there? I'm thinking of 3D Gaussians, but it could be other format support maybe? (I know @lknknm mentioned .ifc at some point)

**Comment by mwestphal** [comment]
Sure! Do you have issue links so I can add them ?

**Comment by mwestphal** [comment]
There is no issue for gaussian splatting, feel free to create one.

About .ifc, it is listed in the format list but its not very high. I dont think we need to list it in the roadmap. If someone does it, we will welcome it of course.

**Comment by Meakk** [comment]
https://github.com/f3d-app/f3d/issues/1253

**Comment by mwestphal** [comment]
Thanks! Ill add that to the roadmap!

**Comment by kathleenhang** [comment]
Some ideas for improving the documentation:
- Add more examples for animations, more specifically, make the distinction more clear for animations. For example, a new user may not understand that .gltf may contain multiple animations within, or even multiple separate animations within a single animation (ex: `interpolationtest.glb`). I think it is also easy to misinterpret "All animations" as keyframes of a single animation. I know I had a lot of trouble understanding and explaining to others what .gltf is capable of. 
- Add a section at the beginning of the developer documentation which shows file paths and download links for each different type of model file of varying types. That way, new developers can quickly play around with F3D instead of searching for these different types of files. I know the gallery section contains all of this, but a "Quickstart Guide" could feel more intuitive.
- Improve the testing section with examples for how to add tests and how they work.

These are some aspects that stuck out to me as I was doing my first contribution for F3D. I guess these should be converted into their own issues?

**Comment by mwestphal** [comment]
> Animations

Indeed, for now there is only this: https://f3d.app/doc/user/USAGE.html#animations
It could be worth it to add a page dedicated to animations or to expand this with more details.

Please open an issue and I will add it to the roadmap :)

> Quick Start guide

It is suposed to be covered by this short section here: https://f3d.app/#quickstart
But I supose we could have a detailed quick start page in the doc with link to resource and such. Please open an issue and I will add it to the roadmap.

> Testing

I couldn't agree more, the testing section is lacking (for reference: https://f3d.app/doc/dev/TESTING.html)
Please open an issue and I will add it to the roadmap.

Thanks for sharing your feedback, it is unvaluable!

**Comment by mwestphal** [comment]
QoL: #1257

**Comment by mwestphal** [comment]
- https://github.com/f3d-app/f3d/issues/1258
 - https://github.com/f3d-app/f3d/issues/1259
 - https://github.com/f3d-app/f3d/issues/1260

**Comment by mwestphal** [comment]
Roadmap has been updated, but comments are welcome!

**Comment by mwestphal** [comment]
2.4.0 has been released, closing
